### PR TITLE
Change start and stop for enable and disable

### PR DESCRIPTION
**Pull request by jparisu**

### CODE REVIEW

**Review by jparisu — COMMENTED** [review]
Changes requested:

- [X] YamlReader for heritable topics
- [x] CommonReader enable must be enable_nts_
- [ ] Track enable check data available
- [x] max history depth value

File: ddspipe_core/src/cpp/core/DdsPipe.cpp [hunk 1]
```diff
@@ -92,113 +101,77 @@ DdsPipe::~DdsPipe()
...
+    allowed_topics_ = allowed_topics;
 
-        // Set new Allowed list
-        allowed_topics_ = allowed_topics;
+    logDebug(DDSROUTER, "New DDS Pipe allowed topics configuration: " << allowed_topics_);
 
-        logDebug(DDSROUTER, "New DDS Pipe allowed topics configuration: " << allowed_topics_);
+    if (!enabled_)
+    {
+        return utils::ReturnCode::RETCODE_OK;
```
Comment (irenebm) [inline ddspipe_core/src/cpp/core/DdsPipe.cpp:122]: NIT: it is more accurate to return RETCODE_NOT_ENABLED

Comment (jparisu) [reply ddspipe_core/src/cpp/core/DdsPipe.cpp:122]: I do not agree. This method is independent of the object state, so calling it being disabled should have a correct behaviour, as calling it being enabled

**Review by irenebm — APPROVED** [review]

**Comment by irenebm** [comment]
LGTM with green CI

**Review by irenebm — APPROVED** [review]

**Pull request merged**